src/component/main-content.js

```diff
@@ -45,17 +30,21 @@ export default class MainContent extends React.Component {
 
   render() {
 
-    let {requests, config, activeWindow} = this.props;
+    let {
+      requestData,
+      config,
+      activeWindow,
+      setFromIndex,
+      filterRequests
+    } = this.props;
+
     let {activeRequest} = this.state;
     let SearchBar, SetupInstructions;
 
-    const filterRequests = (str) => {
-      this._filterRequests(str);
-      this.render();
-    };
-
     const setActiveRequest = (request) => {
-      this._setActiveRequest(request);
+      this.setState({
+        activeRequest: request
+      });
     };
 
     if(activeRequest) {
```

Comment:
Could you check https://github.com/uxebu/james/pull/10/files#diff-ce2b02ac788b14260a1cae97ea32071eR51.  Either you're changing state within `render` or you shadowing a variable name.